### ISSUE TITLE
Issue with Freecad weekly `.AppImage` (black startscreen)

### ISSUE THREAD

**Issue by leoheck** [FreeCAD/FreeCAD-Bundle#97]
Black Start page
![image](https://user-images.githubusercontent.com/1277920/159270350-6ee69ea8-adc1-4a16-9e35-de7505752366.png)

```
OS: Ubuntu Jammy Jellyfish (development branch) (ubuntu:GNOME/ubuntu)
Word size of FreeCAD: 64-bit
Version: 0.20.27936 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 66f3474)
Hash: 66f34746c8e8e2b18f308d705b6a98d9d6942791
Python 3.9.10, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: English/United States (en_US)
Installed mods: 
  * A2plus 0.4.54b
  * fasteners 0.3.38
  * Assembly4 0.11.10
  * kicadStepUpMod 10.11.1
  * ExplodedAssembly
```

**Comment by leoheck** [comment]
weekly, daily, nightly... these things are the same thing to say "dev builds"... but yeah, weekly then..

But these things are not new, expect for the "Start page" thing.

**Comment by leoheck** [comment]
```
OS: Ubuntu Jammy Jellyfish (development branch) (ubuntu:GNOME/ubuntu)
Word size of FreeCAD: 64-bit
Version: 0.20.27936 (Git) AppImage
Build type: Release
Branch: (HEAD detached at 66f3474)
Hash: 66f34746c8e8e2b18f308d705b6a98d9d6942791
Python 3.9.10, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: English/United States (en_US)
Installed mods: 
  * A2plus 0.4.54b
  * fasteners 0.3.38
  * Assembly4 0.11.10
  * kicadStepUpMod 10.11.1
  * ExplodedAssembly
```

**Comment by luzpaz** [comment]
We need more conda packagers looking at this.

**Comment by leoheck** [comment]
@luzpaz is this something that I can make a PR here? Because this looks like a simple thing to solve. 
At least talking about the Image and App Name.

**Comment by luzpaz** [comment]
I imagine the App Name could be fixed here? 
https://github.com/FreeCAD/FreeCAD-Bundle/blob/master/conda/linux/AppDir/freecad_conda.desktop#L3

But how would you differentiate it from other FreeCAD applications that may be installed in parallel?

**Comment by leoheck** [comment]
Check this, please.
https://github.com/FreeCAD/FreeCAD-Bundle/pull/98

**Comment by leoheck** [comment]
I put `Freecad Weekly`. Perhaps we can tweak the logo a bit putting a symbol next to the `Fc`

**Comment by leoheck** [comment]
Freecad Weekly is enough, I guess. No package has this name. But something different in the icon can make this even better.

**Comment by luzpaz** [comment]
Maybe we can take the 'C' out of this, scale it down and superimpose it on the FreeCAD logo? ![](https://camo.githubusercontent.com/fcbeed317b55a6e451772fe75f8894b2e5f5889eba9566b31f78a1215527916f/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f636f6e64612d6465762f636f6e64615f6c6f676f2e737667)

**Comment by leoheck** [comment]
What is conda by the way? How do yo install it? How to do create this package?

I can try to make your idea. I will Update my pr asap. 
Ah, we.can also replace the c of Fc, by this one. It may be good.

**Comment by luzpaz** [comment]
I think @looooo wrote this up on the wiki https://wiki.freecad.org/Conda

Also checkout 
https://github.com/FreeCAD/FreeCAD_Conda

**Comment by leoheck** [comment]
Awesome, thanks for the material.

**Comment by KenwoodFox** [comment]
Also thanks for the material, im coming from another issue with a glider dependency breaking a while ago. I can confirm this issue is still present, tested on latest Arch, Manjaro and KDE Neon (ubuntu)

**Comment by leoheck** [comment]
@luzpaz hey, I have added a logo following your ideia in this PR here, https://github.com/FreeCAD/FreeCAD-Bundle/pull/98
Could you check if it is good? It does not look pretty good for me. 

![image](https://user-images.githubusercontent.com/1277920/161140211-59d81731-c208-45f4-8e70-ec694048a558.png)

**Comment by KenwoodFox** [comment]
Maybe the gear shiuld be replaced with the Conda C idk if it all fits together :P

**Comment by leoheck** [comment]
Something like this?
It feels that is missing something.. haha

![image](https://user-images.githubusercontent.com/1277920/161172736-3d7bf2a4-213e-4712-80b7-a489fcddf33c.png)

**Comment by leoheck** [comment]
It would be better if it has some kind of color gradient matching the other symbols

**Comment by leoheck** [comment]
Maybe something like this, with the C "Bundling" the gear

![image](https://user-images.githubusercontent.com/1277920/161173086-b69d355c-8795-4c48-996e-d7c90624e54a.png)

**Comment by leoheck** [comment]
![image](https://user-images.githubusercontent.com/1277920/161173476-4f1605f4-832c-4417-9360-c29018102007.png)

**Comment by leoheck** [comment]
![image](https://user-images.githubusercontent.com/1277920/161173589-3cd88fec-847a-4fe2-b994-9246c4027806.png)

**Comment by leoheck** [comment]
I though this, but then we are kind of messing with the original symbol, and then it won't be as good as it is. 

Maybe then, having a big C and reducing the gear to fit inside as the C was Bundleing it?

**Comment by KenwoodFox** [comment]
> Something like this? It feels that is missing something.. haha
> 
> ![image](https://user-images.githubusercontent.com/1277920/161172736-3d7bf2a4-213e-4712-80b7-a489fcddf33c.png)

Honestly this is my favorite :3 but all the ones you posted are very hepful

**Comment by leoheck** [comment]
I can change my pr or make a new one with this one. Then we can use  it for some time to see how it feels.

**Comment by luzpaz** [comment]
Looks like the StartWB isn't even loading correctly. I don't see it's toolbar either. Weird thing is that I don't see any errors

```
OS: Manjaro Linux (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.20.28567 (Git) AppImage
Build type: Release
Branch: (HEAD detached at eb65256)
Hash: eb652569c1bc905b9ece89598ce99e2a2549acae
Python 3.9.12, Qt 5.12.9, Coin 4.0.0, OCC 7.5.3
Locale: English/United States (en_US)
Installed mods: 
  * A2plus 0.4.55a
```

**Comment by leoheck** [comment]
StartWB? Is this related with this issue here?

**Comment by luzpaz** [comment]
> StartWB? Is this related with this issue here?

The ticket name says 'Black startscreen'. I think it's StartWB related. If you close the window (ctrl+w) and open a new file (ctrl+n) the black screen goes away.

**Comment by leoheck** [comment]
Ah of course, sorry. I read your reply quickly and then for no reason I was thinking about "star" something...

**Comment by luzpaz** [comment]
No worries :smile:   
TBF, we should separate the ticket in to distinct issues so it's not confusing.

**Comment by luzpaz** [comment]
StartWB relevant thread: https://forum.freecadweb.org/viewtopic.php?p=584953#p584953

**Comment by chennes** [comment]
This looks like three different bug reports: can we split it?

**Comment by onekk** [comment]
From

https://forum.freecadweb.org/viewtopic.php?p=585485#p585485

adding:

```
export QTWEBENGINE_DISABLE_SANDBOX=1
```

Seems to solve the problem.

further investigation seems to lead to:

https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1939993

I don't know if it is relevant.

Regards

Carlo D.

**Comment by luzpaz** [comment]
How does one add entries in to AppRun? 
Does that happen here: https://github.com/FreeCAD/FreeCAD-Bundle/blob/master/conda/linux/create_bundle.sh#L85-L89 

Someone in the forum thread mentioned: 
![Screenshot_20220402_152420](https://user-images.githubusercontent.com/4140247/161398139-7ecbce3b-67e3-4c19-99f8-7a6a1afa4eb4.png)

**Comment by luzpaz** [comment]
Workaround:  
Invoke AppImage from the terminal like this:  
`QTWEBENGINE_DISABLE_SANDBOX=1 ./FreeCAD_weekly-builds-28567-Linux-Conda_glibc2.12-x86_64.AppImage`

**Comment by onekk** [comment]
see maybe:

 https://forum.freecadweb.org/viewtopic.php?p=585715#p585715

For other type of solution.

Regards

Carlo D.

**Comment by luzpaz** [comment]
This is fixed in 7d89de1c4d265ef8b5ccad72672815133a55c913 thanks to @onekk and @looooo and others. Closing ticket

**Comment by luzpaz** [comment]
@looooo can you generate new appimage build to incorporate 7d89de1c4d265ef8b5ccad72672815133a55c913

**Comment by onekk** [comment]
Hello, unrelated question, but miniconda environment is not updated incorporating this patch to the more "general problem" of "Blank (Black) Start Page" on some Systems.

Not a critics, only to note the disalignment between packaging system (AppImage, miniconda, FlatPack ....)

why not make this modification to "stock source tree" that will make things work despite the "Packaging system":

in file:

.../miniconda3/envs/fcenv-dev/Mod/Start/Init.py (This is the location of the miniconda install, I know :-D)

I've inserted these lines:

```
import os
os.environ["QTWEBENGINE_DISABLE_SANDBOX"] = "1"
```
And all it working.

Regards

**Comment by luzpaz** [comment]
@looooo the fix worked in the appimage for me.  So it looks like https://github.com/FreeCAD/FreeCAD-Bundle/commit/7d89de1c4d265ef8b5ccad72672815133a55c913 worked.  
Note: there is a secondary issue that has popped us since, see #102

**Comment by onekk** [comment]
The code has to be applied to the stock freecad sources so it is loaded when Start page is invoked, and will be integrated in all the "packaging system" solution for AppImage patch the AppRun environment only, good thing, but other "packaging system" are not patched so as example my miniconda install was broken when updating the miniconda as it has overwritten my "workaround lines" in /mod/Start/Init.py.

The workaround should be used only in FC as Init.py is modifying only the sys.environment of the running FreeCAD I think, at least in an AppImage or in a miniconda install that is not very far from an AppImage in workings as it appears that an AppImage use a "sort of"  miniconda environment internally.

#102 seems related to some problem with Xorg but Xorg is external to the AppImage, or maybe FC is asking for some OpenGL things that the driver is not supplying, (a rough guess), some other programs, has a "OpenGL" version settings, that could address problems if the OS distribution use an old OpenGL but as usual, I don't know in deep things, about miniconda and AppImage workings.

I suppose that libc++ and Xorg are managed by the "OS Distribution" and miniconda and AppImages use the "OS services", in this case it seems strange that FC is asking for a precise place where libraries are residing, as different distributions have different places where libraries are put.

Debian Ubuntu is known to have different places, at least in the past, now it seems more "standardized", probably diue to the SystemD adoption, but in past some libraries that other distributions placed in /usr/lib or in /usr/share/lib are found in /usr/lib64 or /usr/chare/lib64, I have had some program that need some symlinks as when running on Debian they won't find libraries and fails (notably was a proprietary driver for a printer), then driver suppliers have done a complex installation script that take in account these differences.

But probably here this discussion is rather OT, sorry.

Regards.

Carlo D.

**Comment by looooo** [comment]
@onekk I added a patch to the freecad-feedstock. https://github.com/looooo/freecad-feedstock/blob/master/recipe/qtwebengine_disable_sandbox.patch

If you think this patch needs to be applied to the freecad source please discuss this via an separate issue or in the forum.

**Comment by onekk** [comment]
@looooo  (hope the number o o is correct :-D)

I think that this patch should be made "upstream" i.e directly in FC sources, pleas on what forum I have to discuss it?

Regards

Carlo D.
